### DISCUSSION TITLE
Feature: dropdown to choose from all configured containers not just last ?

### DISCUSSION THREAD

**Discussion by denisimmoos** [mozilla/multi-account-containers#2687] [Ideas]
<img width="864" alt="Screenshot 2024-10-30 at 08 09 13" src="https://github.com/user-attachments/assets/5b02c441-4d11-40c7-9e7b-41f9574aca92">

To improve the user experience and provide more flexibility, a dropdown menu containing all available containers would be a better option than a single button. 
The text for this feature could be:

"Select a container to open this site:
[Dropdown menu with container options]

**Comment by achernyakevich-sc** [comment]
@denisimmoos you could check #2294 that partially represent your proposal. It is more for reference.

In reality since I created that issue the situation changed and now `Remember my decision for this site` checkbox works for the case when people expect to unassign site from the container. But the approach could be very similar. Though items in the selectbox could start from the name of the container (so searching by keyboard will work). Something like:

* `- Use current container -`
* `- Use Private container and do not ask again -`
* `ContainerA - use this container`
* `ContainerB - use this container`
* `Finance - use this container`
* `Private - use this container`

This way we could avoid the case when some people have dozens of containers. We will not need search as container will be sorted in the selectbox (two first are reserved) and could be accessible by starting typing container name. I would not recommend to implement auto-completer component here for searching container as by my feeling 99% of users have  no more than 10 containers. :)

@dannycolin - if this idea will be in general agreed I will try to find somebody who will implement it. I know that this case looks like as misusage of `Alway Open This Site in...` feature but it is really quite convenient for cases when you work in different containers with different accounts of AWS. :)

**Reply by kelimuttu** [reply]
I think this is a brilliant idea @achernyakevich-sc 

@dannycolin what do you think?

**Reply by dannycolin** [reply]
I definitely like the idea of offering more options. UX-wise, I'm not sure I'd include the "do not ask again" inside the select element.

I'm just brainstorming but something like:

* Container A is the assigned container and it has a dedicated button since it's the most likely option.
* Container B is the current container and it is the default value for the dropdown to speed up the selection

```
[ Open in ContainerA ] 

OR open in:

[ Open in ContainerB ]  
[ ------------------ ]
[ Container C        ]
[ Container D        ]

[ ] Remember my decision for this site
```

@achernyakevich-sc What do you think?

I'm also open to any feedback, suggestion on the wording on the page so it makes it more clear what's going to happen when an option is selected.
